src/pal/src/sharedmemory/sharedmemory.cpp

```diff
@@ -1007,6 +1043,14 @@ SIZE_T SharedMemoryManager::s_creationDeletionFileLockOwnerThreadId = SharedMemo
 void SharedMemoryManager::StaticInitialize()
 {
     InitializeCriticalSection(&s_creationDeletionProcessLock);
+
+    s_runtimeTempDirectoryPath = InternalNew<PathCharString>();
+    s_sharedMemoryDirectoryPath = InternalNew<PathCharString>();
+
+    VerifyStringOperation(s_runtimeTempDirectoryPath && s_sharedMemoryDirectoryPath);
```

Comment:
This function cannot throw. Please return error code instead. The shared memory implementation is the only part of PAL that uses exceptions and that needs to stay in its confines.